core/src/main/java/com/google/zxing/oned/Code128Writer.java

```diff
@@ -79,13 +97,33 @@
         case ESCAPE_FNC_2:
         case ESCAPE_FNC_3:
         case ESCAPE_FNC_4:
-          break;
+          continue;
         default:
           if (c > 127) {
             // support for FNC4 isn't implemented, no full Latin-1 character set available at the moment
             throw new IllegalArgumentException("Bad character in input: " + c);
           }
       }
+      switch (forcedCodeSet) {
+        case CODE_CODE_A:
+          if (c > 95) {
+            // the last 32 ASCII characters are not present in code set A
+            throw new IllegalArgumentException("Bad character in input for forced code set A: " + c);
+          }
+          break;
+        case CODE_CODE_B:
+          if (c < 32) {
+            // the first 32 ASCII characters are not present in code set B
+            throw new IllegalArgumentException("Bad character in input for forced code set B: " + c);
```

Comment:
For this case, maybe others, do you want to include the ASCII value of the char? c is a `char` and it will be unprintable in this case